kolibri/plugins/learn/assets/src/views/EmbeddedSidePanel/index.vue

```diff
@@ -316,13 +316,19 @@
       handleActivity(activity) {
         let learning_activities;
         if (activity) {
-          learning_activities = {
-            [activity]: true,
-          };
+          if (this.value.learning_activities[activity]) {
+            delete this.value.learning_activities[activity];
+            this.$emit('input', { learning_activities });
```

Comment:
This needs `...this.value` - I think you've overcomplicated the diff here, and directly deleting from the value is not what we should do for the v-model compatibility.

This funcion should instead looking something like this:
```
      handleActivity(activity) {
        let learning_activities;
        if (activity || !this.value.learning_activities[activity]) {
          learning_activities = {
            [activity]: true,
          };
        } else {
          learning_activities = {};
        }
        this.$emit('input', { ...this.value, learning_activities });
      },
```

Reply:
I originally had tried something like this, but it doesn't seem to work... I'll take another look now

Reply:
OK - I did test it locally and confirmed that the current implementation did cause _all_ search terms to get cleared when a learning activity is deselected.